bool req_1
bool req_2
bool req_3